lib/types.bzl

```diff
@@ -149,4 +163,5 @@ types = struct(
     is_function = _is_function,
     is_depset = _is_depset,
     is_set = _is_set,
+    is_partial = _is_partial,
```

Comment:
Can we move this function to the partial file so that types doesn't depend on it?

Reply:
I'm not sure what you mean by "so that types doesn't depend on it". In my current changes, `is_partial()` in `types.bzl` does not depend on `partial.bzl`, just as `is_set()` does not depend on any of the skylib set implementations.  Both of these predicates do duplicate knowledge of how their respective data structures are implemented elsewhere.

Reply:
Exactly: the knowledge belongs in sets or partial, not in types. Even though there may be no physical dependency, there is a logical one, and it is unnecessary.